org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/callhierarchy/ShowFilterDialogAction.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.ui.callhierarchy;
+
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.window.Window;
+
+import org.eclipse.ui.PlatformUI;
+
+import org.eclipse.jdt.internal.ui.IJavaHelpContextIds;
+import org.eclipse.jdt.internal.ui.JavaPluginImages;
+
+class ShowFilterDialogAction extends Action {
```

Comment:
On a second thought, rename this one to `ShowCallHierarchyFilterDialogAction`. The reason is that by moving this class out of its previous container class (`CallHierarchyFiltersActionGroup`) the name of the action is too generic (if you look for the **type** (`Ctrl` + `Shift` + `t`) `ShowFilterDialogAction` then you'll see that there are other 2 internal classes with that name.